airbyte-webapp/src/pages/connections/CreateConnectionPage/usePreloadData.ts

```diff
@@ -0,0 +1,113 @@
+import { useEffect } from "react";
+import { useLocation, useNavigate, useSearchParams } from "react-router-dom";
+
+import {
+  DestinationDefinitionRead,
+  DestinationRead,
+  SourceDefinitionRead,
+  SourceRead,
+} from "core/request/AirbyteClient";
+import { useGetDestination } from "hooks/services/useDestinationHook";
+import { useGetSource } from "hooks/services/useSourceHook";
+import { useDestinationDefinition } from "services/connector/DestinationDefinitionService";
+import { useSourceDefinition } from "services/connector/SourceDefinitionService";
+
+export function hasSourceId(state: unknown): state is { sourceId: string } {
```

Comment:
[airbyte-webapp/src/components/ui/Text/TextWithOverflowTooltip.tsx](https://github.com/airbytehq/airbyte/pull/22701#discussion_r1107747584)

Reply:
[TextWithOverflowTooltip.tsx](https://github.com/airbytehq/airbyte/pull/22701#discussion_r1120975697)? 😂 

For future readers, KC is linking to a question about improving the type signature for `state` in the original PR in the `airbyte` repo; in response, I'm linking to my answer, which is that "better" type signatures conflict with the actual `state` value, which is `location.state`, which is a web API with a predefined type of `unknown`, which is a type which exists specifically to refuse to match anything helpful.